webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -4100,6 +4101,116 @@ const firstVisibleMsg = {
                 }
                 $mck_msg_to.focus();
             };
+
+            _this.isWithinBusinessHours = function (team) {
+                // don't show message if business hour setting are not set
+                if (!team || !team.businessHourMap || !team.timezone) {
+                    return true;
+                }
+                const convertToMinutes = (timeStr) => {
+                    const hours = parseInt(timeStr.slice(0, 2), 10);
+                    const minutes = parseInt(timeStr.slice(2), 10);
+                    return hours * 60 + minutes;
+                };
+
+                const parseTimezoneOffset = (timezone) => {
```

Comment:
why is the timezone thing required here? is it the same way it is handled at other places such as showing the time of the messages?

Reply:
In other parts of the system, we directly show the local time based on the user's or agent's location. For instance, if you're an agent, you'll see your local time, and if you're a user, you'll see your local time, regardless of where you're located globally.

However, when it comes to business hours, there's an additional check required. If a user sends a message, we need to verify whether the time falls within the agent's business hours, which may differ from the user's local time. That's why I've used time timezone handling here.